### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by omarkhan**
This pull request updates the instance manager documentation:

- Explain what the instance manager does
- List the configuration options and explain what they mean
- Add example configuration
- Explain how to set up an ssh key pair for sandbox provisioning
- Explain how to configure openstack images and flavours
- Show how to manage sandboxes from the shell

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
...
+      --name IMAGE_NAME \
+      --progress
+
+where `IMAGE_NAME` is the name used to identify the image in OpenStack. The
+`OPENSTACK_SANDBOX_BASE_IMAGE` setting in your `.env` file should match this
+name:
+
+    OPENSTACK_SANDBOX_BASE_IMAGE='{"name": "IMAGE_NAME"}'
+
+### OpenStack flavours
```
Comment (pomegranited) [inline README.md]: @omarkhan Bitten by British English..  I know that's really how flavours is spelled, but it should probably be in American like the rest of the document :)

File: README.md [hunk 2]
```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
 server or the integration tests.
 
 The environment variables in `.env` customize the settings from
-`opencraft/settings.py` which are loaded via `env()`.
+`opencraft/settings.py` which are loaded via `env()`. For more information about
+each setting, see [`opencraft/settings.py`](opencraft/settings.py). As a
```
Comment (pomegranited) [inline README.md]: @omarkhan It's worth referencing the excellent settings documentation you created too, e.g.:

 "For more information about each setting, see [settings.py], or the [Application Settings](#application-settings) section below."

File: README.md [hunk 3]
```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
...
+    OPENSTACK_SANDBOX_FLAVOR='{"name": "m1.medium"}'
+
+### Application settings
+
+* `DEBUG` Turn on debug mode. Use in development only (default: False)
+* `SECRET_KEY` Set this to something unique and keep it secret (required)
+* `DATABASE_URL` Your database, e.g. `postgres://localhost/opencraft` (required)
+* `REDIS_URL` (default: `redis://localhost:6379/`)
+* `HUEY_ALWAYS_EAGER` Don't run tasks in the background. Use in development only
+  (default: False)
```
Comment (pomegranited) [inline README.md]: @omarkhan nitpicking: "Don't run tasks in the background" sounds like an instruction?  Maybe "Set to True to run tasks in the foreground.  Use in development only (default: False)."

File: README.md [hunk 4]
```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
...
+  connect to sandbox instances (default: `opencraft`). This key pair should be
+  [registered with OpenStack](http://docs.openstack.org/user-guide/cli_nova_configure_access_security_for_instances.html)
+  first, and should be the default ssh key for the user the instance manager
+  runs as.
+* `OPENSTACK_SANDBOX_SSH_USERNAME` The user to run ansible playbooks as when
+  provisioning the sandbox (default: `ubuntu`)
+* `INSTANCE_EPHEMERAL_DATABASES` By default, instances use local mysql and mongo
+  databases. Set this to False to use external databases instead (default: True)
+* `INSTANCE_MYSQL_URL` If using an external mysql database, set it's url here
+* `INSTANCE_MONGO_URL` If using an external mongo database, set it's url here
```
Comment (pomegranited) [inline README.md]: @omarkhan nitpicking: should be *its* not *it's*

File: README.md [hunk 5]
```diff
@@ -89,36 +240,33 @@ Migrations
...
+In order to login to the development server locally you will need to create a
+superuser by running:
 
-```
-honcho run ./manage.py createsuperuser
-```
+    honcho run ./manage.py createsuperuser
+
+Once created, you will be able to login to with the username and password you
+set up.
```
Comment (pomegranited) [inline README.md]: @omarkhan Seriously nitpicking:  "you will be able to login *to* with the username.." should be "you will be able to login with the username.."

File: README.md [hunk 6]
```diff
@@ -128,89 +276,69 @@ Then go to:
...
 
+    make run WORKERS=2
 
 
 Process description
 -------------------
 
-This runs three processus via Honcho, which reads `Procfile` or `Procfile.dev` and loads the
-environment from the `.env` file:
+This runs three processus via Honcho, which reads `Procfile` or `Procfile.dev`
```
Comment (pomegranited) [inline README.md]: @omarkhan sp: "processes" not "processus"

File: README.md [hunk 7]
```diff
@@ -218,19 +346,49 @@ Debug
...
-```
-$ make shell
+    make shell
+
+
+Provisioning sandboxes
+----------------------
+
+When configured correctly, the instance manager will automatically provision
+sandboxes whenever a pull request is made on GitHub. If you want to provision a
```
Comment (pomegranited) [inline README.md]: @omarkhan "whenever a pull request is made on GitHub".. append "on your configured ``WATCH_FORK``, by members of your configured ``WATCH_ORGANIZATION``" ?

**Comment by pomegranited** [comment]
@omarkhan Your updates look great!  Your writing is very clear.

Let me know when you're done, and I'll review the rest.  From looking at OC-1018, the only missing bits are:
* shell console manual reprovisioning
* shell console manual instance delete

And I was a bit confused by the `User workflow` section..  If you have time, could you perhaps give an example PR description that demonstrates the ephemeral vs persistent databases switch "on the same line as the sandbox domain in the pull request body"?  Are there any other magic ways to configure the automatically created instance?  I'm asking for this mainly so I know how to use our Instance Manager.

Cheers!

File: README.md [hunk 8]
```diff
@@ -81,7 +87,152 @@ credentials for third-party services manually in order to run the development
...
+```
+
+### A note on SSH keys
+
+The instance manager uses [Ansible](https://www.ansible.com/) to provision
+openedx sandboxes. Ansible uses SSH to run commands on remote servers, so you
+will need to configure the instance manager with an SSH key pair to use for this
+purpose.
+
+First, **as the user that the instance manager runs as**, run:
```
Comment (pomegranited) [inline README.md]: @omarkhan Does this mean I also need to create a new user inside my vagrant instance to run the instance manager, or is one already created for me?  If not, that should be part of the vagrant setup instructions (with the local install setup, it's clear enough already).

**Comment by omarkhan** [comment]
Thanks for the review @pomegranited! I have updated the pull request based on your feedback, can you take another look?

**Comment by pomegranited** [comment]
Looks great @omarkhan !  :+1:

**Comment by omarkhan** [comment]
Thanks @pomegranited! I have squashed the commits, just waiting for the tests to pass so I can merge

**Pull request merged**